pages/_app.js

```diff
@@ -27,37 +26,45 @@ export default function App({ Component, pageProps }) {
     return;
   }
 
-  if (initialTransactions && !transactions.length) {
-    setTransactions(initialTransactions);
-  }
-  // Until here
+  async function handleAddTransaction(newTransaction) {
+    const response = await fetch("/api/transactions", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newTransaction),
+    });
+
+    if (!response.ok) {
+      console.error(response.status);
+      return;
+    }
 
-  function handleAddTransaction(newTransaction) {
-    setTransactions([{ _id: uid(), ...newTransaction }, ...transactions]);
+    mutate();
   }
-  function handleEditTransaction(updatedTransaction, id) {
-    const updatedTransactions = transactions.map((transaction) =>
-      transaction._id == id
-        ? {
-            ...transaction,
-            name: updatedTransaction.name,
-            amount: updatedTransaction.amount,
-            currency: updatedTransaction.currency,
-            date: updatedTransaction.date,
-            description: updatedTransaction.description,
-            category: updatedTransaction.category,
-            paymentMethod: updatedTransaction.paymentMethod,
-            direction: updatedTransaction.direction,
-          }
-        : transaction
-    );
-    setTransactions(updatedTransactions);
+
+  async function handleEditTransaction(updatedTransaction, id) {
+    const response = await fetch(`/api/transactions/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(updatedTransaction),
+    });
+
+    if (response.ok) {
+      mutate();
+    }
```

Comment:
in the other function you do an inverse check. Please choose one way and keep it consistent.